src/app/static/src/app/components/plots/bubble/BubblePlot.vue

```diff
@@ -0,0 +1,186 @@
+                </l-circle-marker>
+            </template>
+            <map-control :initialDetail=detail
+                         :show-indicators="false"
+                         @detail-changed="onDetailChange"></map-control>
+        </l-map>
+    </div>
+</template>
+
+<script lang="ts">
+    import Vue from "vue";
+    import {Feature} from "geojson";
+    import {LGeoJson, LMap, LCircleMarker, LTooltip} from "vue2-leaflet";
+    import MapControl from "../MapControl.vue";
+    import {GeoJSON, Layer} from "leaflet";
+    import {ChoroplethIndicatorMetadata} from "../../../generated";
+    import {getFeatureIndicators, toIndicatorNameLookup} from "./utils";
+    import {BubbleIndicatorValuesDict, Dict, LevelLabel} from "../../../types";
+
+    interface Props {
+        features: Feature[],
+        featureLevels: LevelLabel[]
+        indicators: ChoroplethIndicatorMetadata[],
+        chartdata: any[]
+    }
+
+    interface Data {
+        style: any,
+        //TODO: persist these as part of selections
+        colorIndicator: string,
+        sizeIndicator: string,
+        detail: number
+    }
+
+    interface Methods {
+        updateBounds: () => void,
+        getRadius: (feature: Feature) => number,
+        getColor: (feature: Feature) => string,
+        getTooltip: (feature: Feature) => string,
+        onDetailChange: (newVal: number) => void
+    }
+
+    interface Computed {
+        featureIndicators: Dict<BubbleIndicatorValuesDict>,
+        featuresByLevel: { [k: number]: Feature[] },
+        currentFeatures: Feature[],
+        maxLevel: number,
+        indicatorNameLookup: Dict<string>
+    }
+
+    const props = {
+        features: {
+            type: Array
+        },
+        featureLevels: {
+            type: Array
+        },
+        indicators: {
+            type: Array
+        },
+        chartdata: {
+            type: Array
+        }
+    };
+
+    export default Vue.extend<Data, Methods, Computed, Props>({
+        name: "BubblePlot",
+        components: {
+            LMap,
+            LGeoJson,
+            LCircleMarker,
+            LTooltip,
+            MapControl
+        },
+        props: props,
+        data(): Data {
+            return {
+                style: {
+                    weight: 1,
+                    fillOpacity: 1.0,
+                    color: 'grey',
+                    fillColor: 'rgb(200,200,200)'
+                },
+                //TODO: initialise these from metadata
+                colorIndicator: "prevalence",
+                sizeIndicator: "plhiv",
+                detail: 0
+            }
+        },
+        computed: {
+            featureIndicators() {
+                return getFeatureIndicators(
+                    this.chartdata,
+                    this.currentFeatures,
+                    this.indicators,
+                    10, //min radius in pixels (TODO: should come from metadata)
+                    10000 //max radius
```

Comment:
This is an absurd radius because the max value of the indicators is so high compared to any actual values (for the default indicators we're using). To be replaced by more sensible metadata in due course..

Reply:
This is going to be a real problem. 

I was a little confused at first at how low the numbers were in the tooltip - e.g. up to a maximum of 50 people living with HIV in a district! Then I realised that because there's no filtering, it's taking the last value for each age/sex category, which is men over the age of 80. When you look at both sexes between 15 - 49, the number of PLHIV in the same area will be literally 3 orders of magnitude larger, up to around 60k.

 The fact that the max value here has to be 10,000 is a red flag as obviously a circle of radius 10,000px is impossible. Realistically I think 50px might be a max radius? 

It's the homogenous colour problem but in this case I think it's more severe. The only way I can think of making this workable is if the min and max values are re-calculated for every filter combination, so bubble scale will represent something different each time the filters change. I would think bubble min and max radius should remain fixed (and I think it makes sense for us to fix these rather than expect them in metadata, since they're determined by what is realistic to display rather than anything data specific?)

In the case of the colour scale we were reluctant to change the meaning of the colour scale based on selected filters, and I think that is legit. In the case of the bubbles, I just don't see any other way, and I think it's also slightly less problematic?

Reply:
Agreed. Think this actually makes things easier, or at least less prone to go horribly wrong...

Reply:
OK, actually I've been looking at putting in something slightly different to what you suggest, in that I'm calculating the actual min and max values across all the data, not for every combination of filters. This feels a bit more natural as it means you tend to get your biggest bubbles at country/region level for wide age ranges etc, and smallest bubbles at district level/narrow age ranges. We'll make the legend show the actual numeric range we're working with, and these can stay fixed for the output data set. Does that sound ok?

Reply:
Maybe I'll ask Jeff..

Reply:
That seems like it would be better but does it work visually? I think the lowest no PLHIV at the smallest level of regional detail was 1 person, vs the highest no PLHIV at the same level being around 60k.

Reply:
And overall wouldn't the range then be something like 1 person - several million people?

Reply:
I can see the benefit of both approaches. Good to keep a fixed scale sometimes, good to be able to see the full range in a single view sometimes. Maybe it ends up as a toggle in the UI which you want to use... I've almost got the filters working in another branch, I'll plug on with my current approach for the moment and merge it with the filters branch to see what it looks like with different filter combos..